libs-haskell/da-hs-base/src/DA/Service/Logger/Impl/GCP.hs

```diff
@@ -178,25 +179,33 @@ withGcpLogger
     :: GCPConfig
     -> (Lgr.Priority -> Bool) -- ^ if this is true log to GCP
     -> Lgr.Handle IO
-    -> (GCPState -> Lgr.Handle IO -> IO a)
+    -> (Maybe GCPState -> Lgr.Handle IO -> IO a)
     -- ^ We give access to both the GCPState as well as the modified logger
     -- since the GCPState can be useful to bypass the message filter, e.g.,
     -- for metadata messages which have info prio.
     -> IO a
 withGcpLogger config p hnd f = do
-    gcpState <- initialiseGcpState config hnd
-    let logger = hnd
-            { Lgr.logJson = newLogJson gcpState
-            }
-    let worker = forever $ mask_ $ do
-            -- We mask to avoid messages getting lost.
-            entry <- atomically (readTChan (gcpLogChan gcpState))
-            sendLogs gcpState [entry]
-    (withAsync worker $ \_ -> do
-        f gcpState logger) `finally` do
-        logs <- drainChan (gcpLogChan gcpState)
-        sendLogs gcpState logs
-        Lgr.logJson hnd Lgr.Info ("Flushed " <> show (length logs) <> " logs")
+    gcpStateM <- initialiseGcpState config hnd
+    case gcpStateM of
+      Just gcpState -> do
+        isOptedIn <- doesFileExist (optedInFile gcpState)
+        isOptedOut <- doesFileExist (optedOutFile gcpState)
+        if isOptedIn && not isOptedOut
```

Comment:
Afaict, this check is going to fail on an empty setup. You enable telemetry in DAML Studio, we run `damlc ide --telemetry=yes`, we get here, the files are not created and we don’t enable telemetry.

We need to keep the logic to write the file if we are opted in.

Reply:
I think you're right. I moved the logic back to the assistant.